File: ts/easy/mergeFish.ts
const seaArr = ['cod', 'herring', 'marling'];
const freshArr = ['asp', 'carp', 'ide', 'trout'];

function merge(sea: string[], fresh: string[]): string[] {
    const result: string[] = [];
    const copySea = [...sea];
    const copyFresh = [...fresh];
    const MOST_HEAVY_STR = 'zzzzzz';

    while (copySea.length || copyFresh.length) {
        const seaFish = copySea[0] || MOST_HEAVY_STR;
        const freshFish = copyFresh[0] || MOST_HEAVY_STR;

        if (seaFish < freshFish) {
            const fish = copySea.shift();
            if (fish) {
                result.push(fish);
            }
        } else if (seaFish > freshFish) {
            const fish = copyFresh.shift();
            if (fish) {
                result.push(fish);
            }
        }
    }
    return result;
}
console.log('Output: ', merge(seaArr, freshArr));

File: python/easy/mergeFish.py
from typing import List

sea_list = ["cod", "herring", "marling"]
fresh_list = ["asp", "carp", "ide", "trout"]

def merge(sea: List[str], fresh: List[str]) -> List[str]:
    result = []
    copy_sea = sea[::]
    copy_fresh = fresh[::]
    MOST_HEAVY_STR = 'zzzzzz'

    while len(copy_sea) > 0 or len(copy_fresh) > 0:
        sea_fish = MOST_HEAVY_STR
        fresh_fish = MOST_HEAVY_STR

        if len(copy_sea):
            sea_fish = copy_sea[0]
        if len(copy_fresh):
            fresh_fish = copy_fresh[0]

        if sea_fish < fresh_fish:
            fish = copy_sea.pop(0)
            if fish:
                result.append(fish)
        elif sea_fish > fresh_fish:
            fish = copy_fresh.pop(0)
            if fish:
                result.append(fish)
    return result
print(merge(sea_list, fresh_list))
